aws/resource_aws_launch_template.go

```diff
@@ -335,9 +335,20 @@ func resourceAwsLaunchTemplate() *schema.Resource {
 							Optional: true,
 						},
 						"ipv4_addresses": {
-							Type:     schema.TypeSet,
+							Type:     schema.TypeList,
 							Optional: true,
-							Elem:     &schema.Schema{Type: schema.TypeString},
```

Comment:
This is a breaking change too for existing configurations since it changes the resource configuration format. We'll need to implement this in a manner where both the old and new configurations are supported at the same time (e.g. a new attribute). If we need to go down the new attribute route, we have some documentation available here (its for renaming an attribute but the required steps are similar): https://www.terraform.io/docs/extend/best-practices/deprecations.html#renaming-an-optional-attribute

We really should have an acceptance test configuration that covers the old behavior and ensures it continues to work with the updates.